test/platform/target/network/scan_result_test.exs

```diff
@@ -0,0 +1,38 @@
+defmodule Farmbot.Target.Network.ScanResultTest do
+  use ExUnit.Case, async: false
+
+  alias Farmbot.Target.Network.ScanResult
+
+  describe "decode/1" do
+    test "returns a an empty struct when we pass in an empty map" do
+      result = ScanResult.decode(%{})
+      expected = %Farmbot.Target.Network.ScanResult{
+                                                     bssid: nil,
+                                                     capabilities: nil,
+                                                     flags: nil,
+                                                     level: nil,
+                                                     noise: nil,
+                                                     security: nil,
+                                                     ssid: nil
+                                                   }
+      assert result == expected
```

Comment:
I don't think that it is intended behaviour to test that passing in an empty map should return an empty struct.